example-client/README.md

```diff
@@ -3,11 +3,219 @@ Example CoAP client
+
+<img src="images/example-running.png" width="600">
+
+    *note: replace ${ENDPOINT_NAME} and ${ACCOUNT_ID} with the value defined in the Configuring section below*
+
+*${ENDPOINT_NAME} can be obtained by running the following command*
+```
+keytool -list -v -storepass secret -keystore example-client-device-keystore.jks 2> /dev/null | grep 'Alias name: mbed-cloud'
+```
+
+<img src="images/example-connected.png" width="600">
+
+# Building with Maven
+
+#### Clone java-coap
+
+```
+git clone https://github.com/ARMmbed/java-coap
+```
+
+### Change to the project
+
+```
+cd java-coap
+```
+
+### Maven install the project
+
+```
+mvn package install -DskipTests
+```
+
+### Change to the example client
+
+```
+cd example-client
+```
+
+# Configuring
+
+## Generating Keystore, Certificates, Security Keys
+
+### Set these variables which can be retrieved from https://portal.us-east-1.mbedcloud.com/account
+
+```
+ACCOUNT_ID=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+```
+
+<img src="images/account.png" width="600">
+
+### Set these variables which can be retrieved from https://portal.us-east-1.mbedcloud.com/access/keys
+
+```
+API_KEY=ak_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+```
+
+<img src="images/apikey.png" width="600">
+
+### Use uuidgen tool to generate a unique endpoint name
+
+```
+ENDPOINT_NAME=$(uuidgen)
+```
+
+### Make a directory to store your keystore, certificates, and security keys
+
+```
+mkdir keystore && cd keystore
+```
+
+### Issue a REST call via cURL command to get the LWM2M server certificate from the Mbed Cloud endpoint (lwm2m-east-1.mbedcloud.com) extract the server cert and dump it to a PEM file
+
+Note: you can also cut and paste this from the Portal
+
+```
+curl -s -X GET \
+  https://api.us-east-1.mbedcloud.com/v3/server-credentials/lwm2m \
+  -H 'Authorization: Bearer '"${API_KEY}"'' \
+  -H 'Cache-Control: no-cache' \
+  -H 'Content-Type: application/json' > | jq -r '."server_certificate"' > lwm2m_server_ca_certificate.pem
+```
+
+<img src="images/server_cert.png" width="600">
+
+### Personal Self Signed Certificate Authority
+
+### Create private key for the LWM2M CA certificate
+
+* [Properties for manually generated keys and certificates](https://cloud.mbed.com/docs/current/provisioning-process/index.html)
+
+* [Setting up your own certificate authority](https://cloud.mbed.com/docs/v1.2/provisioning-process/setting-up-your-own-certificate-authority.html)
+
+```
+openssl ecparam -out my_ca_private_key.pem -name prime256v1 -genkey
+```
+
+### Sign a CA with our private key and the CA will be uploaded to mbedcloud
+
+```
+openssl req -key my_ca_private_key.pem -new -sha256 -x509 -days 12775 -out my_ca_public_cert.pem -subj /CN=CA -config <(echo '[req]'; echo 'distinguished_name=dn'; echo '[dn]'; echo '[ext]'; echo 'basicConstraints=CA:TRUE') -extensions ext
+```
+## LWM2M Device Certificates
+
+### Create device private key
+
```

Comment:
Again, why not having numbered steps instead of headings with very little content below?